temporalio/docker-compose
# add explicit persistent volumes to containers

## What was changed
Docker containers for ElasticSearch and Databases are now explicitly specify named volumes.

## Why?
when Docker re-creates containers from images, ElasticSearch data is lost, because container re-creation deletes volumes not explicitly specified in dockerfile. container re-creation can happen for variety of reasons, for example when one changes docker-compose.yml definition to expose ports. 

Postgres or MySQL or Cassandra data isn't lost, because image definition for those contain explicit VOLUME command (eg "VOLUME /var/lib/postgresql/data" in https://github.com/docker-library/postgres/blob/master/13/alpine/Dockerfile, line 154), however moving it to docker-compose definition for clarity as well. 

so now under volumes tab in Docker Desktop, or in the volumes list under 'docker volumes list' command, there is a named volume for data, such as 'temporal_db-data' and 'temporal_es-data', as opposed to randomly named volume for database data and no volume for ElasticSearch data.

## Checklist


1. Closes - (no issue, ad-hoc find)

2. How was this tested:
before the change:
 - run 'docker compose up'
 - run some test workflows
 - verify they both present in elastic search ('tctl workflow list') as well as in the database ('tctl workflow show wf_id')
 - stop server/containers, modify container definiton in docker-compose file, e.g. by adding ports section
 - run 'docker compose up' again and notice that container is 'recreated'
 - verify that elastic search data is lost, but database data is here

after the change:
 - same steps, but after container recreation, data is not lost.

3. Any docs updates needed?
 - presumably, this was not the most common scenario, otherwise it would've reported earlier. however it is also an assumed behavior - if database data is not lost, ElasticSearch shouldn't be either...
 - README file change is unrelated, but just keeping it up-to-date
